### PR TITLE
Implement logic to inherit the new configuration from parts of the ex…

### PR DESCRIPTION
**Pull request by stmh**
hi @d34dman 

this merge request implements what we discussed on friday. It will allow us to mark certain keys in the templates to inherit from existing configuration. 

Instead of `#` i am using `__` as we used that for other placeholders and `#` marks the key as a comment. 

The precedence is as discussed: Value in es.yml > existing configuration > template.yml

I changed the templates only for `settings` and `third_party_settings` for now.

best,
Stephan

### CODE REVIEW

**Comment by d34dman** [comment]
Did a small refactoring,

First,
- Instead of using `__` prefix, i switched to use the same strategy as `__CHILD_REF__` is handled.
- So the prefix is `__|`, Eg. `__|settings`, `__|third_party_settings`
- Advantage is, the prefix doesn't have to be strictly 2 character long, with an exception for `__CHILD_REF__`'s case
- Also allowing, us the possibility to use it for more strategies, and prefixes

Second,
I wanted to pass the decision or control on replacement/reuse to templates too, Just in case.


Third,
Updated the baseline to adjust for the order of the output, so test passes.

Fourth,
I thought about introducing ksort before the array comparison is made while testing. What do you think?

**Comment by d34dman** [comment]
(sorry, closed by mistake).

File: src/transformers/Utils/PlaceholderService.php [hunk 1]
```diff
@@ -15,6 +15,13 @@ class PlaceholderService
 
     protected const CHILD_REFERENCE = '__CHILD_REF__';
 
+    // es.yml > existing configuration
+    const STRATEGY_OVERRIDE_FROM_EXISTING = '_$';
```
Comment (d34dman) [inline src/transformers/Utils/PlaceholderService.php:19]: STRATEGY_OVERRIDE_FROM_EXISTING  This has not been implemented yet.

**Comment by stmh** [comment]
H @d34dman 

There was a reason I used a two-pass approach to  save all keys with encoded strategy and remove them from the array before handling the actual array values.

As forEach works on a copy of an array, it will still step over any unset entries. It depends on the order of the keys, if existing values win, or actual values from the entity-definition.yml from es. That means: any overrides by the es.yml files wont be reflected anymore. 

Try this yourself, edit one of the properties in the content-block yml and run the test. They will pass, as the value of your content-block.es wont get affected anymore.

We need to add some tests for this scenario, too.

**Comment by stmh** [comment]
And can you elaborate about the use case for STRATEGY_OVERRIDE_FROM_EXISTING?

**Comment by stmh** [comment]
@d34dman I added a test which is green with my initial changes and fails with your refactoring.

**Comment by d34dman** [comment]
Thanks for the failing tests and explanations.
I will come up with an update later today. I had suspected that some values we wanted to carry forward from es, while some we wanted to merge if available and some we wanted it to be left untouched in the destination config files. For this, instead of handling the logic in Class methods, I wanted to expose this to templates. Secondly, the 'meaning' of what was being done was not clear to me.

I will spend some time later today to study this further. Worst case, we can undo what i did :)

**Comment by d34dman** [comment]
Update
I have implemented the replacement strategies as discussed earlier today. Also mapped STRATEGY_INHERIT_FROM_EXISTING to STRATEGY_EDT (i.e Es > Drupal > Template).

Seems like this covers the test case you had added, but fails/causes regression in old tests. I need to evaluate if those where faulty test baselines which needs to be adapted or failure in template files.

Also while at it, I might remove STRATEGY_INHERIT_FROM_EXISTING, i.e. `__|` prefix and replace it with `__edt|` prefix in template files.

**Comment by stmh** [comment]
whats the current state of this MR?

**Comment by d34dman** [comment]
Got sidetracked and forgot about this. 
The regressions reported in my earlier comment are actually failing test case. Couldn't come up with updating the templates to use the correct strategy right away when I realised what was happening.

TODO: Update template to use the correct strategies for key's.

**Comment by stmh** [comment]
will work on this today

**Comment by stmh** [comment]
Hi @d34dman 
found the underlying issue, all tests are green. Please review my changes and feel free to merge it back.

**Pull request merged**